federated_aws_cli/cache.py

```diff
@@ -107,22 +155,24 @@ def read_aws_shared_credentials():
 
 
 @_requires_safe_cache_dir
-def write_aws_shared_credentials(credentials):
-    # Create a sha256 of the role arn, so fix length and remove weird chars
+def write_aws_shared_credentials(credentials, role_arn, role_map=None):
     path = os.path.join(DOT_DIR, "credentials")
```

Comment:
Why set this variable, it looks like it isn't used? Oh, this isn't in the PR.

Reply:
?????

Reply:
It looks like `path` is set but not used.

Reply:
It's used twice in the function, once at `with _safe_write(path) as f:` and once with `return path`.

Reply:
(I'm pretty sure that flake8 would fail if there was an unused variable, but I could be wrong?)